src/xrdsvc/HttpReplicaMgtModule.cc

```diff
@@ -0,0 +1,356 @@
+#include "XrdSsi/XrdSsiCluster.hh"
+
+// Qserv headers
+#include "http/Exceptions.h"
+#include "http/RequestBody.h"
+#include "http/RequestQuery.h"
+#include "mysql/MySqlUtils.h"
+#include "util/IterableFormatter.h"
+#include "wconfig/WorkerConfig.h"
+#include "wcontrol/Foreman.h"
+#include "wcontrol/ResourceMonitor.h"
+#include "wpublish/ChunkInventory.h"
+#include "xrdsvc/SsiProvider.h"
+#include "xrdsvc/XrdName.h"
+
+extern XrdSsiProvider* XrdSsiProviderLookup;
+
+using namespace std;
+using json = nlohmann::json;
+
+namespace {
+// These markers if reported in the extended error response object of the failed
+// requests could be used by a caller for refining the completion status
+// of the corresponding Controller-side operation.
+json const extErrorInvalidParam = json::object({{"invalid_param", 1}});
+json const extErrorReplicaInUse = json::object({{"in_use", 1}});
+
+string makeResource(string const& database, int chunk) { return "/chk/" + database + "/" + to_string(chunk); }
+
+string vec2str(vector<string> const& v) {
+    ostringstream ss;
+    ss << lsst::qserv::util::printable(v, "", "", ",");
+    return ss.str();
+}
+
+}  // namespace
+
+namespace lsst::qserv::xrdsvc {
+
+void HttpReplicaMgtModule::process(string const& context, shared_ptr<wcontrol::Foreman> const& foreman,
+                                   shared_ptr<qhttp::Request> const& req,
+                                   shared_ptr<qhttp::Response> const& resp, string const& subModuleName,
+                                   http::AuthType const authType) {
+    HttpReplicaMgtModule module(context, foreman, req, resp);
+    module.execute(subModuleName, authType);
+}
+
+HttpReplicaMgtModule::HttpReplicaMgtModule(string const& context,
+                                           shared_ptr<wcontrol::Foreman> const& foreman,
+                                           shared_ptr<qhttp::Request> const& req,
+                                           shared_ptr<qhttp::Response> const& resp)
+        : HttpModule(context, foreman, req, resp),
+          _providerServer(dynamic_cast<xrdsvc::SsiProviderServer*>(XrdSsiProviderLookup)),
+          _clusterManager(_providerServer->GetClusterManager()),
+          _dataContext(_clusterManager->DataContext()) {}
+
+json HttpReplicaMgtModule::executeImpl(string const& subModuleName) {
+    string const func = string(__func__) + "[sub-module='" + subModuleName + "']";
+    debug(func);
+    enforceInstanceId(func, wconfig::WorkerConfig::instance()->replicationInstanceId());
+    enforceWorkerId(func);
+    if (subModuleName == "GET")
+        return _getReplicas();
+    else if (subModuleName == "SET")
+        return _setReplicas();
+    else if (subModuleName == "ADD")
+        return _addReplica();
+    else if (subModuleName == "REMOVE")
+        return _removeReplica();
+    else if (subModuleName == "REBUILD")
+        return _rebuildInventory();
+    throw invalid_argument(context() + func + " unsupported sub-module");
```

Comment:
I think it's probably worth making and enum with to and from string functions for subModuleName and a switch statement here.

Reply:
I too thought about this. passing `enum` would be safer here than passing strings. The problem is that I have many modules (not just this one). So, I will migrate those at once in a separate refactoring code clean-up effort. I will register it in my TODO list.